pontoon/contributors/utils.py

```diff
@@ -252,16 +272,16 @@ def _get_12_month_average(monthly_rates):
                 action_type=ActionLog.ActionType.TRANSLATION_CREATED,
                 translation__date=F("translation__approved_date"),
             )
-        )
+        ),
     )
 
-    approval_rates = _get_monthly_rates(peer_approvals, peer_rejections)
-    approval_rates_12_month_avg = _get_12_month_average(approval_rates)
-    self_approval_rates = _get_monthly_rates(peer_approvals, self_approvals)
-    self_approval_rates_12_month_avg = _get_12_month_average(self_approval_rates)
+    approval_rates = get_shares_of_totals(peer_approvals, peer_rejections)
+    approval_rates_12_month_avg = get_sublist_averages(approval_rates, 12)
+    self_approval_rates = get_shares_of_totals(self_approvals, peer_approvals)
```

Comment:
Phew, thanks to writing the unit tests, I found a bug. Self-approval rates were previously (line 260) actually calculated as peer approval rates.